ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -139,22 +142,34 @@ class JITModBuilder {
 
       // Iterate over all instructions within this block and copy them over
       // to our new module.
+      // FIXME It would be nice to merge the two skipping mechanisms
+      // (ExtCallInst/last_call).
       for (auto I = BB->begin(); I != BB->end(); I++) {
-        // Skip calls to debug intrinsics (e.g. @llvm.dbg.value). We don't
-        // currently handle debug info and these "pseudo-calls" cause our blocks
-        // to be prematurely terminated.
-        if (isa<DbgInfoIntrinsic>(I))
-          continue;
+        // If we've returned from an external call, skip ahead to the
+        // instruction where we left off.
+        if (ExtCallInst != nullptr) {
+          if (&*I != ExtCallInst) {
+            continue;
+          } else {
+            ExtCallInst = nullptr;
+          }
+        }
 
-        // If we've returned from a call, skip ahead to the instruction where
-        // we left off.
+        // If we've returned from an internal call, skip ahead to the
+        // instruction where we left off.
         if (last_call != nullptr) {
           if (&*I == last_call) {
             last_call = nullptr;
```

Comment:
Silly question perhaps: does this deal with arbitrary levels of "if we've returned from an internal call" due to inlining?

Reply:
Yes. Lukas implemented that. See the fib test.

Reply:
The thing that worries me is that `last_call` suggests we can only handle this to one level. Perhaps that's all we need to handle, but it's not obvious to me.

Reply:
`last_call` is populated by popping from a stack of inlined calls.

But @ptersilie is your guy for this logic. I'm still a little over-whelmed by it!

Reply:
Can you file an issue / talk to @ptersilie about this? I agree that it's not something for this PR, but it does look fishy, and I don't want us to forget to look at it.

Reply:
`last_call` is populated here:
```
if (isa<ReturnInst>(I)) {
  last_call = inlined_calls.back();
}
```
where `inlined_calls` is a vector of calls we've followed (and the tracer has inlined).